In [1]:
from TheoryStatsMath.BruteForce import asymetric_random_walk_time_limit

In [2]:
asymetric_random_walk_time_limit(5, 10, 0, 1.3, 2, 0.62, 10)

(3866397, 2902037, 0.5712395215791423)

asymetric_random_walk_time_limit(5, 10, 0, 1.3, 2, 0.62, 10)
- (3825175, 2869352, 0.5713883893514807)
- (3635804, 2725898, 0.5715143526056392)
- (3359532, 2527159, 0.5706995661909212)
- 26209346 19494228  0.5734638170747872 -- Rust 100 sec
- 27027059 20115492  0.5733049745229103 -- Rust 100 sec
-  2580093  1918893  0.5734832248866745 -- rust 100 sec
-  2564247  1907288  0.5734601205178982   -- rust 10 sec

In [5]:
from TheoryStatsMath.BruteForce import asymetric_random_walk_run_limit
asymetric_random_walk_run_limit(5, 10, 0, 1.3, 2, 0.62, 40_000_000)

(22851344, 17148656, 40000000, 0.5712836)

asymetric_random_walk_run_limit(5, 10, 0, 1.3, 2, 0.62, 10_000_000)

| UB hits | LB hits | total runs | percentage | run time |
| -------- | ------| ------------| -----------| -------- |
|5712589  | 4287411 | 10000000 | 0.5712589 | 38.5s      |
| 5711720 | 4288280 | 10000000  |  0.571172 |  NA   |
| 11427821| 8572179 | 20000000  |  0.57139105 | 52.9 |
| 11424076| 8575924 | 20000000 |   0.5712038 | 48.9s |
| 22851344 |  17148656 | 40000000|  0.5712836| 2m 13.6s |
|22848303 | 17151697 | 40000000 | 0.571207575 | mp 1m 26,6s |
|22845302| 17154698| 40000000| 0.57113255 |mp 1m 26,6s |
|22846589| 17153411| 40000000| 0.571164725 |mp 1m 26,6s |
|22853802| 17146198| 40000000| 0.57134505 |mp 1m 26,6s |
|22846257| 17153743| 40000000| 0.571156425 |mp 1m 26,6s |
 |45693130| 34306870| 80000000| 0.571164125| mp 2m 46.3s |
|45691407| 34308593| 80000000| 0.5711425875|mp 2m 46.3s |
|45702063| 34297937| 80000000| 0.5712757875|mp 2m 46.3s |
|45703958| 34296042| 80000000| 0.571299475|mp 2m 46.3s |
|45702188| 34297812| 80000000| 0.57127735|mp 2m 46.3s |
|91388376| 68611624| 160000000| 0.57117735| NA |
|91404740| 68595260| 160000000| 0.571279625| NA |
|91408934| 68591066| 160000000| 0.5713058375| NA |
|91403747| 68596253| 160000000| 0.57127341875| NA |
|91387688| 68612312| 160000000| 0.57117305| NA |
 |182800886| 137199114| 320000000| 0.57125276875| 11m 42.2s |
|182815469| 137184531| 320000000| 0.571298340625| 11m 42.2s |
|182810399| 137189601| 320000000| 0.571282496875| 11m 42.2s |
|182804807| 137195193| 320000000| 0.571265021875|11m 42.2s |
|182815106| 137184894| 320000000| 0.57129720625| 11m 42.2s |


Rust results with f32

RunResult { upper_hits: 183481958, lower_hits: 136518042, total_runs: 320000000, percentage: 0.5733811 }    
RunResult { upper_hits: 183494105, lower_hits: 136505895, total_runs: 320000000, percentage: 0.5734191 }    
RunResult { upper_hits: 183499699, lower_hits: 136500301, total_runs: 320000000, percentage: 0.57343656 }   
RunResult { upper_hits: 183479505, lower_hits: 136520495, total_runs: 320000000, percentage: 0.57337344 }   
RunResult { upper_hits: 183488958, lower_hits: 136511042, total_runs: 320000000, percentage: 0.573403 }

Rust results with f64

RunResult { upper_hits: 182794582, lower_hits: 137205418, total_runs: 320000000, percentage: 0.57123306875 }    
RunResult { upper_hits: 182813460, lower_hits: 137186540, total_runs: 320000000, percentage: 0.5712920625 }     
RunResult { upper_hits: 182813127, lower_hits: 137186873, total_runs: 320000000, percentage: 0.571291021875 }   
RunResult { upper_hits: 182809316, lower_hits: 137190684, total_runs: 320000000, percentage: 0.5712791125 }     
RunResult { upper_hits: 182796659, lower_hits: 137203341, total_runs: 320000000, percentage: 0.571239559375 }   

In [1]:
import multiprocessing
from TheoryStatsMath.BruteForce import asymetric_random_walk_run_limit


PROCESSES = 5
with multiprocessing.Pool(PROCESSES) as pool:
    params = [(1, 3, 0, 1, 1, 0.6, 320_000_000) for x in range(PROCESSES)]
    results = [pool.apply_async(asymetric_random_walk_run_limit, p) for p in params]

    for r in results:
        print('\t', r.get())

	 (151579972, 168420028, 320000000, 0.4736874125)
	 (151566391, 168433609, 320000000, 0.473644971875)
	 (151560400, 168439600, 320000000, 0.47362625)
	 (151588731, 168411269, 320000000, 0.473714784375)
	 (151580131, 168419869, 320000000, 0.473687909375)
